### PR TITLE
Adding short commit to `version_info`

### PR DESCRIPTION
**Pull request by sydney-runkle**
## Change Summary

Adding most recent `commit` info to the result of the `version_info` call.

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [x] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by sydney-runkle** [comment]
Please review

File: pydantic/version.py [hunk 1]
```diff
@@ -41,6 +44,12 @@ def version_info() -> str:
         if name in package_names:
             related_packages.append(f'{name}-{dist.version}')
 
+    pydantic_dir = os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
+    most_recent_commit = (
+        git.git_revision(pydantic_dir) if git.is_git_repo(pydantic_dir) and git.have_git() else 'unknown'
+    )
+    del pydantic_dir
```
Comment (hramezani) [inline pydantic/version.py:51]: Why do we need this?

File: tests/test_version.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 def test_version_info():
     s = version_info()
     assert re.match(' *pydantic version: ', s)
-    assert s.count('\n') == 6
+    assert s.count('\n') == 7
```
Comment (hramezani) [inline tests/test_version.py:14]: probably we can check the existence of `commit`. `assert re.match(' *commit: ', s)`

**Review by hramezani — APPROVED** [review]

File: tests/test_version.py (hunk 2 above)
Comment (sydney-runkle) [reply tests/test_version.py:14]: Yeah. I suppose we should do this for the other fields as well.

File: tests/test_version.py [hunk 3]
```diff
@@ -9,9 +9,21 @@
...
+        'platform',
+        'related packages',
+        'commit',
+    ]
+
     s = version_info()
-    assert re.match(' *pydantic version: ', s)
-    assert s.count('\n') == 6
+    for field in version_info_fields:
+        assert re.search(f'{field}', s)
```
Comment (hramezani) [inline tests/test_version.py:25]:
```suggestion
    assert all([f'{field}:' in s for field in version_info_fields])
```

Comment (sydney-runkle) [reply tests/test_version.py:25]: Yeah that's cleaner, thanks

**Pull request merged**